Std/Data/List/Basic.lean

```diff
@@ -361,7 +361,7 @@ drop_while (· != 1) [0, 1, 2, 3] = [1, 2, 3]
   | a :: l, n => bif p a then n else go l (n + 1)
 
 /-- Returns the index of the first element equal to `a`, or the length of the list otherwise. -/
-def indexOf [BEq α] (a : α) : List α → Nat := findIdx (a == ·)
+def indexOf [BEq α] (a : α) : List α → Nat := findIdx (· == a)
```

Comment:
This needs some justification. I think there was some discussion earlier about switching one or more of these which resulted in a TODO comment in mathlib, but I forget the details.

Reply:
I've updated the PR description.

I couldn't find the TODO in mathlib, unfortunately. Mathlib is consistent however about always using `· == a`, so maybe we should just do that globally here too.